### ISSUE TITLE
Add Intersection over Union (IoU) metric

### ISSUE THREAD

**Issue by jshin1394** [google/metrax#60]

**Comment by jshin1394** [comment]
Using the Keras [implementation](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/IoU) as basis, add the metric to metrax.
